core/config/registry_accessor.hpp

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_CORE_CONFIG_REGISTRY_ACCESSOR_HPP_
+#define GKO_CORE_CONFIG_REGISTRY_ACCESSOR_HPP_
+
+
+#include <string>
+
+
+#include <ginkgo/core/config/registry.hpp>
+
+
+namespace gko {
+namespace config {
+namespace detail {
+
+
+class registry_accessor {
```

Comment:
I use it as the registry friend to access the data without giving the access to user.
By using friend function (if you mean make it a namespace), user still can call this function.
User can not use this class because the definition is in internal.

Reply:
Oh, right, then it needs to be a struct/class.